tests/test_extension_registry.py

```
@@ -0,0 +1,1632 @@
+        self.assertFalse(results['success'])
+        self.assertEqual(results['unknown_imported'], 0)
+        self.assertTrue(any('Could not resolve suggested category' in error for error in results['errors']))
+        self.assertEqual(len(self.manager.get_unknown_extensions()), 0)
+
+    def test_import_resolves_foreign_keys_by_name(self):
+        """Test that import resolves foreign keys using natural keys when IDs differ."""
+        existing_category_id = self.manager.create_category("Existing Category", "Original", 1, True)
+        self.manager.create_category("Secondary Category", "Secondary", 2, True)
+        existing_platform_id = self._create_test_platform("Existing Platform")
+
+        self.export_file = "/tmp/test_foreign_key_import.json"
```

Ruff (0.12.2)

Probable insecure usage of temporary file or directory: "/tmp/test_foreign_key_import.json"

(S108)